pkg/registry/core/service/storage/rest.go

```diff
@@ -242,18 +256,6 @@ func (rs *REST) Delete(ctx context.Context, id string, deleteValidation rest.Val
 
 	svc := obj.(*api.Service)
 
-	// Only perform the cleanup if this is a non-dryrun deletion
-	if !dryrun.IsDryRun(options.DryRun) {
-		// TODO: can leave dangling endpoints, and potentially return incorrect
-		// endpoints if a new service is created with the same name
```

Comment:
does this no longer applies?

Reply:
Good catch, this is probably still true if we encounter unexpected error when deleting the endpoint. One thing that is not clear in my mind is whether or not the deletion of service will be completed if we hit such error in the afterDeleteHook, and whether or not a new service with the same name will be able to be created if the prior service deletion has not been completed. (The comments was added [a long while back](https://github.com/jennybuckley/kubernetes/commit/3ffe259ac773f5c9ec91d6e8b5751020a05af56b) and I suspect if it is stale.)

Nevertheless, this PR focuses more on moving the de-allocation step as is, so let me copy over the comments for now.